spring-integration-file/src/test/java/org/springframework/integration/file/tail/FileTailingMessageProducerTests.java

```diff
@@ -90,6 +88,29 @@ public void testApache() throws Exception {
 		testGuts(adapter, "tailer");
 	}
 
+    @Test
+    @TailAvailable
+    public void canRecalculateCommandWhenFileOrOptionsChanged() throws IOException {
+        File firstFile = File.createTempFile("first", ".txt");
+        String firstOptions = "-f options";
+        File secondFile = File.createTempFile("second", ".txt");
+        String secondOptions = "-f newoptions";
+        OSDelegatingFileTailingMessageProducer adapter = new OSDelegatingFileTailingMessageProducer();
+        adapter.setFile(firstFile);
+        adapter.setOptions(firstOptions);
+
+        adapter.setOutputChannel(new QueueChannel());
+        adapter.setTailAttemptsDelay(500);
+        adapter.setBeanFactory(mock(BeanFactory.class));
+        adapter.afterPropertiesSet();
+
+        assertEquals("tail " + firstOptions + " " + firstFile.getAbsolutePath(), adapter.getCommand());
+        adapter.setFile(secondFile);
+        assertEquals("tail " + firstOptions + " " + secondFile.getAbsolutePath(), adapter.getCommand());
+        adapter.setOptions(secondOptions);
+        assertEquals("tail " + secondOptions + " " + secondFile.getAbsolutePath(), adapter.getCommand());
```

Comment:
Can't we have this `reconfiguration` test as a real one? Similar to that `testOS()`